### PR TITLE
Add API Response Model in ChargeDialogeFragment

### PR DESCRIPTION
**Pull request by mayank-kgp**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by mayank-kgp** [comment]
@therajanmaurya  I have added POJO in ChargeDialogeFragment.
Please review it when you get some time.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 1]
```diff
@@ -44,41 +38,47 @@
...
-    Spinner sp_charge_name;
+    Spinner spChargeName;
 
     @BindView(R.id.amount_due_charge)
-    EditText et_amout_due;
+    EditText etAmoutDue;
 
     @BindView(R.id.et_date)
-    EditText charge_due_date;
+    EditText chargeDueDate;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Field name should be like et_chargeDueDate

Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Field name should be like et_chargeDueDate

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 2]
```diff
@@ -44,41 +38,47 @@
...
 
     @BindView(R.id.et_charge_locale)
-    EditText charge_locale;
+    EditText chargeLocale;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Field name should be like et_ chargeLocale

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 3]
```diff
@@ -44,41 +38,47 @@
...
 
     @BindView(R.id.bt_save_charge)
-    Button bt_save_charge;
+    Button btSaveCharge;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Field name should be like btn_saveCharge

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 4]
```diff
@@ -44,41 +38,47 @@
...
 
     @Inject
     ChargeDialogPresenter mChargeDialogPresenter;
 
+    private List<String> chargeNameList = new ArrayList<>();
+
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: remove extra lines gape

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 5]
```diff
@@ -121,21 +121,22 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         inflateChargesSpinner();
+        inflateChargeNameSpinner();
 
-        duedateString = charge_due_date.getText().toString();
+        duedateString = chargeDueDate.getText().toString();
         duedateString = DateHelper.getDateAsStringUsedForCollectionSheetPayload(duedateString)
                 .replace("-", " ");
 
-        bt_save_charge.setOnClickListener(new View.OnClickListener() {
+        btSaveCharge.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: We are using ButterKhife, Remove the OnClickListener implementation and use ButterKhife @OnClick annotation

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java (hunk 3 above)
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: field name should not contain underscores, this field name is perfectly fine

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java (hunk 2 above)
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: field name should not contain underscores, this field name is perfectly fine

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java (hunk 1 above)
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: field name should not contain underscores, this field name is perfectly fine

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java (hunk 3 above)
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Ok, One more thing, we should add prefix or not like btn, sp, et?

**Comment by mayank-kgp** [comment]
@therajanmaurya I have done the changes. Please let me know if anything more is need to be updated. :)

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 6]
```diff
@@ -159,8 +155,8 @@ private void initiateChargesCreation(ChargesPayload chargesPayload) {
 
     public void inflatedueDate() {
         mfDatePicker = MFDatePicker.newInsance(this);
-        charge_due_date.setText(MFDatePicker.getDatePickedAsString());
-        charge_due_date.setOnClickListener(new View.OnClickListener() {
+        etChargeDueDate.setText(MFDatePicker.getDatePickedAsString());
+        etChargeDueDate.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Remove OnClickListener Implementation and Use ButterKnife ```@OnClick``` annotation

**Comment by therajanmaurya** [comment]
@mayank-kgp Make above changes and Let @ishan1604 know to give final review and merge.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 7]
```diff
@@ -118,33 +111,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         ButterKnife.bind(this, rootView);
         mChargeDialogPresenter.attachView(this);
-
         inflatedueDate();
         inflateChargesSpinner();
+        inflateChargeNameSpinner();
+        return rootView;
+    }
 
-        duedateString = charge_due_date.getText().toString();
```
Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: @therajanmaurya Please review this change.

**Comment by mayank-kgp** [comment]
@therajanmaurya  I have done the required changes.
@ishan1604 Please have a final look.

**Comment by mayank-kgp** [comment]
@ishan1604 Please review it.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 8]
```diff
@@ -25,63 +24,57 @@
...
-    EditText charge_due_date;
+    EditText etChargeDueDate;
 
     @BindView(R.id.et_charge_locale)
-    EditText charge_locale;
+    EditText etChargeLocale;
 
     @BindView(R.id.bt_save_charge)
-    Button bt_save_charge;
+    Button btSaveCharge;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: change ```bt --> btn```

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 9]
```diff
@@ -118,33 +111,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
-                chargesPayload.setChargeId(Id);
-                initiateChargesCreation(chargesPayload);
-            }
-        });
         return rootView;
     }
 
+    @OnClick(R.id.bt_save_charge)
+    public void saveNewCharge() {
+
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: No need for this extra line

**Comment by mayank-kgp** [comment]
@therajanmaurya I have done required changes. Please review it.

**Comment by mayank-kgp** [comment]
@ishan1604 @therajanmaurya  I have done the change in commit message.
Please have a look. :)

**Pull request merged**